Ensure SEGIntegration initializes with Adobe

ladanazita opened the pull request:
This is what I believe was originally wrong with my implementation after end to end testing.

The original issue surfaced was after configuring Adobe with sample app in the repo, I set a break point on the [createWithProperties](https://github.com/segment-integrations/analytics-ios-integration-adobe-analytics/blob/master/Pod/Classes/SEGAdobeIntegration.m#L43) method, to ensure that this was being called. The tests were setup to validate against the ADBMediaObject mock, but the e2e tests failed to hit that break point, telling me my actual logic doesn't not reach that point.  

E2E Test setup:

```
#import <Segment-Adobe-Analytics/SEGAdobeIntegrationFactory.h>

@implementation SEGAdobeAppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
    SEGAnalyticsConfiguration *configuration = [SEGAnalyticsConfiguration configurationWithWriteKey:@"<my write key>"];
    configuration.trackApplicationLifecycleEvents = NO; // Enable this to record certain application events automatically!
    configuration.recordScreenViews = NO; // Enable this to record screen views automatically!
    [configuration use:[SEGAdobeIntegrationFactory instance]];
    [SEGAnalytics setupWithConfiguration:configuration];
    [SEGAnalytics debug:YES];

    [[SEGAnalytics sharedAnalytics] track:@"Video Playback Started" properties:@{
                                                                                 @"content_asset_id" : @"1234",
                                                                                 @"ad_type" : @"pre-roll",
                                                                                 @"video_player" : @"youtube",
                                                                                 } options: @{}];
    return YES;
}

```


I've implemented a solution that solves for not properly passing in the Adobe heartbeat instances (ADBObject, ADBMediaHeartbeat) when SEGAdobeIntegration factory was being created by passing in the mediheartbeat and mediaobject factories to SEGAdobeIntegrationFactory.m

Let me know if I'm way off here!

ladanazita commented:
This works for now, but I need to refactor this
* Can have one factory AdobeObjectFactory build both ADBMediaHeartbeat and ADBMediaObject 
* Move into it's own file
* Have the podspec not include downloading this file to make it private

f2prateek commented on Pod/Classes/SEGAdobeIntegration.h:
```diff
@@ -40,7 +40,7 @@
-- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings;
-- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings andADBMobile:(id _Nullable)ADBMobile andADBMediaHeartbeatFactory:(id<SEGADBMediaHeartbeatFactory> _Nullable)ADBMediaHeartbeatFactory andADBMediaHeartbeatConfig:(ADBMediaHeartbeatConfig *_Nullable)config andADBMediaObjectFactory:(id<SEGADBMediaObjectFactory> _Nullable)ADBMediaObjectFactory;
+//- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings;
+- (instancetype _Nonnull)initWithSettings:(NSDictionary *_Nonnull)settings andADBMobile:(id _Nullable)ADBMobileClass andADBMediaHeartbeat:(ADBMediaHeartbeat *_Nullable)ADBMediaHeartbeat andADBMediaHeartbeatFactory:(id<SEGADBMediaHeartbeatFactory> _Nullable)ADBMediaHeartbeatFactory andADBMediaHeartbeatConfig:(ADBMediaHeartbeatConfig *_Nullable)config andADBMediaObject:(ADBMediaObject *_Nullable)me ...
```
Why is this passing both `ADBMediaHeartbeat` and `ADBMediaHeartbeatFactory`? I understand you need one for tests and for the real implementation, but the factory should handle both cases.

f2prateek commented:
> Can have one factory AdobeObjectFactory build both ADBMediaHeartbeat and ADBMediaObject
Not required, separate is fine.

> Move into it's own file
Also not required, same file is fine.

> Have the podspec not include downloading this file to make it private
What do you mean here? Which file? There are no podspec changes in this PR.

f2prateek commented on Pod/Classes/SEGAdobeIntegration.m:
```diff
@@ -460,6 +466,41 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
+
+    // Video Ad Break Started/Completed are not spec'd. For now, will document for Adobe and
+    // write a proposal to add this to the Video Spec
+    if ([payload.event isEqualToString:@"Video Ad Break Started"]) {
```
Think ad events should be added in their own PR. The PR is about initializing Adobe correctly - sending ad events seems like a different task.

ladanazita replied:
That change was already merged, I had this against the ad events branch to continue working, so Github just sees that as part of the change

f2prateek commented on Pod/Classes/SEGAdobeIntegration.m:
```diff
@@ -478,16 +519,18 @@ - (ADBMediaHeartbeatConfig *)createConfig:(SEGTrackPayload *)payload
         debugEnabled = true;
     }
 
+    //    self.config = [[ADBMediaHeartbeatConfig alloc] init];
```
Does leaving this commented like this serve a purpose?

ladanazita replied:
oh just missed this

ladanazita commented:
@f2prateek  Made your requested changes, you were right the factory handled passing in the ADBMediaHeartbeat/Object, so I removed that. 

I also cleaned up the naming like you suggested

f2prateek commented on Example/Tests/Tests.m:
```diff
@@ -50,7 +50,7 @@ - (ADBMediaObject *)createWithProperties:(NSDictionary *)properties andEventType
     beforeEach(^{
         mockADBMobile = mockClass([ADBMobile class]);
-        integration = [[SEGAdobeIntegration alloc] initWithSettings:@{} andADBMobile:mockADBMobile andADBMediaHeartbeatFactory:nil andADBMediaHeartbeatConfig:nil andADBMediaObjectFactory:nil];
+        integration = [[SEGAdobeIntegration alloc] initWithSettings:@{} adobe:mockADBMobile andMediaHeartbeatFactory:nil andMediaHeartbeatConfig:nil andMediaObjectFactory:nil];
```
Doesn't really look like this is used by most of the tests. It should be scoped better so we're only doing this for the tests where we need it.

Alternatively we can just pass in the mock factories here .
